### DISCUSSION TITLE
error using dockerslim build

### DISCUSSION THREAD

**Discussion by anujshdodo** [slimtoolkit/slim#628] [Q&A]
Hello,
while trying to use dockerslim to reduce an image size, the build terminates with below error, this shows it complaints about invalid port "0", but where do i change this port ?

level=error msg=terminating error="API error (500): driver failed programming external connectivity on endpoint slimk_3178_20240108093920 (7ed227ce34b5c1d37415f6498d39f64001a1597d97e632064498150d11c15b):  (iptables failed: iptables --wait -t nat -A DOCKER -p tcp -d 0/0 --dport 16418 -j DNAT --to-destination 1xx.17.0.2:0 ! -i docker0: iptables v1.4.21: Port `0' not valid\n\nTry `iptables -h'

**Comment by kcq** [comment]
@anujshdodo this looks like a more fundamental problem... Usually the port number is declared in the Dockerfile for your image with an EXPOSE instruction. You can check either by looking in the Dockerfile for your image if it's your image and you have the original Dockerfile. Alternatively you can use the "xray" command in Slim, which will generate a reverse engineered Dockerfile for you. And you can also use the "history" command in Docker to look for the EXPOSE instruction there.
